sklearn/neighbors/setup.py

```diff
@@ -31,6 +31,11 @@ def configuration(parent_package='', top_path=None):
                          sources=['typedefs.pyx'],
                          include_dirs=[numpy.get_include()],
                          libraries=libraries)
+    config.add_extension("quad_tree",
+                         sources=["quad_tree.pyx"],
+                         include_dirs=[numpy.get_include()],
+                         libraries=libraries,
+                         extra_compile_args=["-O3"])
```

Comment:
Can you add a comment here on why specifically you use a "-O3" flag?

Does it work on all compilers?